src/search/landmarks/exploration.cc

```diff
@@ -147,124 +122,91 @@ void Exploration::build_unary_operators(const OperatorProxy &op) {
 */
 void Exploration::setup_exploration_queue(
     const State &state, const vector<FactPair> &excluded_props,
-    const unordered_set<int> &excluded_op_ids, bool use_h_max) {
+    const unordered_set<int> &excluded_op_ids) {
     prop_queue.clear();
 
     for (size_t var_id = 0; var_id < propositions.size(); ++var_id) {
         for (size_t value = 0; value < propositions[var_id].size(); ++value) {
             ExProposition &prop = propositions[var_id][value];
-            prop.h_add_cost = -1;
+            prop.excluded = false;
             prop.h_max_cost = -1;
-            prop.depth = -1;
-            prop.marked = false;
         }
     }
 
     for (const FactPair &fact : excluded_props) {
         ExProposition &prop = propositions[fact.var][fact.value];
-        prop.h_add_cost = -2;
+        prop.excluded = true;
     }
 
     // Deal with current state.
     for (FactProxy fact : state) {
         ExProposition *init_prop = &propositions[fact.get_variable().get_id()][fact.get_value()];
-        enqueue_if_necessary(init_prop, 0, 0, 0, use_h_max);
+        enqueue_if_necessary(init_prop, 0, 0);
     }
 
     // Initialize operator data, deal with precondition-free operators/axioms.
     for (ExUnaryOperator &op : unary_operators) {
+        op.excluded = false; // Reset from previous exploration
         op.unsatisfied_preconditions = op.precondition.size();
-        if (op.effect->h_add_cost == -2
+        if (op.effect->excluded
             || excluded_op_ids.count(op.op_or_axiom_id)) {
-            op.h_add_cost = -2; // operator will not be applied during relaxed exploration
+            // operator will not be applied during relaxed exploration
+            op.excluded = true;
             continue;
         }
-        op.h_add_cost = op.base_cost; // will be increased by precondition costs
         op.h_max_cost = op.base_cost;
-        op.depth = -1;
 
         if (op.unsatisfied_preconditions == 0) {
-            op.depth = 0;
-            int depth = op.is_induced_by_axiom(task_proxy) ? 0 : 1;
-            enqueue_if_necessary(op.effect, op.base_cost, depth, &op, use_h_max);
+            enqueue_if_necessary(op.effect, op.base_cost, &op);
```

Comment:
I think the operator costs (`base_cost`) can also go away in the next round.

Reply:
Will probably happen when we replace `h_max_cost` by a boolean `reached` in issue1045.